### PR TITLE
Fixes and improvements to Crypto Callbacks and STM32 RNG performance

### PR DESCRIPTION
**Pull request by dgarske**
* Added crypto callback HMAC support.
* Added crypto callback support for ASN `CalcHashId`.
* Added arg checking to cryptocb functions.
* Fix to ensure hash devCtx is cleared.
* Fix for ensuring devId is passed into symmetric init.
* Added flags build option to hashing algorithms.
* Fix for new random seed crypto callback to properly reset error code in NOT_COMPILED_IN case.
* Improvements to the STM32L4 random generation code for improved performance and error handling. Added new `WOLFSSL_STM32_RNG_NOLIB` define to support generic STM32 series RNG without external ST library.

### CODE REVIEW

**Comment by dgarske** [comment]
@toddouska : This is ready for review.

File: src/tls.c [hunk 1]
```diff
@@ -1112,6 +1112,9 @@ static int Hmac_UpdateFinal(Hmac* hmac, byte* digest, const byte* in,
             if (ret != 0)
                 break;
         }
+        /* call final to cleanup */
+        if (ret == 0)
+            ret = wc_HmacFinal(hmac, dummy);
```
Comment (toddouska) [inline src/tls.c]: This needs to be tested for time changes.  Sean, can you review?

File: wolfssl/wolfcrypt/md5.h [hunk 2]
```diff
@@ -114,6 +117,11 @@ WOLFSSL_API int  wc_Md5Copy(wc_Md5*, wc_Md5*);
 WOLFSSL_API void wc_Md5SizeSet(wc_Md5* md5, word32 len);
 #endif
 
+#if defined(WOLFSSL_HASH_FLAGS) || defined(WOLF_CRYPTO_CB)
+    WOLFSSL_API int wc_Md5SetFlags(wc_Md5* md5, word32 flags);
```
Comment (toddouska) [inline wolfssl/wolfcrypt/md5.h]: Do these need to be public?  In what cases should they be called by the user and how?  How do we make sure they're called correctly in the future?

File: src/tls.c (hunk 1 above)
Comment (dgarske) [reply src/tls.c]: @SparkiDev. Please let me know your thoughts for timing impact. The final is important for the CryptoCB case for cleanup

File: wolfssl/wolfcrypt/md5.h (hunk 2 above)
Comment (dgarske) [reply wolfssl/wolfcrypt/md5.h]: @toddouska: Very good point. I’ll make these local (private)

File: src/tls.c (hunk 1 above)
Comment (SparkiDev) [reply src/tls.c]: So, introducing a final does cause timing issues.
I'm not sure what the asm code is doing but if you make the function not inline (take off the static keyword) then it looks fine again.

Comment (dgarske) [reply src/tls.c]: Timing was impacted, so opted for a better cleanup solution using `wc_HmacFree`.

File: wolfssl/wolfcrypt/md5.h (hunk 2 above)
Comment (dgarske) [reply wolfssl/wolfcrypt/md5.h]: @toddouska : I made these hash flag API's private.

**Pull request merged**